lsp/src/uri0.ml

```diff
@@ -24,6 +24,8 @@ let yojson_of_t t = `String (to_string t)
 
 let equal = ( = )
 
+let compare (x : t) (y : t) = compare x y
```

Comment:
Why do you need to define this?

Reply:
Many functor signature require a comparison function. It's usually a nice to have for that reason.

Reply:
I was asking because compare seems to already be in scope, but maybe it's coming from an `open` statement? Hard to know with the diff

Reply:
I'll use `Poly.compare` to be more clear.